1-exercises/D-array-filter/exercise.js

```diff
@@ -8,7 +8,7 @@
 
 let pairsByIndexRaw = [[0, 3], [1, 2], [2, 1], null, [1], false, "whoops"];
 
-let pairsByIndex; // Complete this statement
+let pairsByIndex = pairsByIndexRaw.filter(x => !!x && pairsByIndexRaw.splice(3,4)); // Complete this statement
```

Comment:
This works, well done.
Code should ideally avoid hard coded values like` (3,4)` above, as it is inflexible and prone to break when values/variables it depends on changes.
The filter function should

1. Check that the element is an array ( see [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray) )
2. Check the length of the array is 2

Reply:
Will try and change that